gen3-client/g3cmd/download-multiple.go

```diff
@@ -17,7 +19,110 @@ import (
 	"github.com/uc-cdis/gen3-client/gen3-client/jwt"
 )
 
-func batchDownload(numParallel int, reqs []*grab.Request) {
+func lastString(ss []string) string {
+	return ss[len(ss)-1]
+}
+
+func processOriginalFilename(downloadPath string, actualFilename string) string {
+	_, err := os.Stat(downloadPath + actualFilename)
+	if os.IsNotExist(err) {
+		return actualFilename
+	}
+	extension := filepath.Ext(actualFilename)
+	filename := strings.TrimSuffix(actualFilename, extension)
+	counter := 2
+	for {
+		newFilename := filename + "_" + strconv.Itoa(counter) + extension
+		_, err := os.Stat(downloadPath + newFilename)
+		if os.IsNotExist(err) {
+			return newFilename
+		}
+		counter++
+	}
+}
+
+func processS3URLForFilename(presignedURL string, guid string, downloadPath string, filenameFormat string, overwrite bool, renamedFiles *[]RenamedFileInfo) string {
+	if filenameFormat != "guid" {
+		urlWithFilename := strings.Split(presignedURL, "?")[0]
+		actualFilename := lastString(strings.Split(urlWithFilename, guid+"/"))
+		if actualFilename != "" {
+			if filenameFormat == "original" {
+				if !overwrite {
+					newFilename := processOriginalFilename(downloadPath, actualFilename)
+					if actualFilename != newFilename {
+						*renamedFiles = append(*renamedFiles, RenamedFileInfo{GUID: guid, OldFilename: actualFilename, NewFilename: newFilename})
+					}
+					return newFilename
+				}
+				return actualFilename
```

Comment:
this is not a big deal, but you could reduce the max indentation of the code and make it easier to read by reorganizing maybe like this:
```
if filenameFormat == "guid"
  return guid
if filenameFormat == "original"
  if overwrite
    return actualFilename
  newFilename := [...]
  return newFilename
// otherwise filenameFormat == "combined":
return combined filename
```